probe/appclient/multi_client.go

```diff
@@ -100,9 +102,7 @@ func (c *multiClient) Set(hostname string, endpoints []string) {
 	hostIDs := report.MakeIDList()
 	for tuple := range clients {
 		hostIDs = hostIDs.Add(tuple.ID)
-
-		_, ok := c.clients[tuple.ID]
-		if !ok {
```

Comment:
It's only used for Pipes, which shouldn't be triggered if there are no controls.

I could do:

```go
                if _, ok := c.clients[tuple.ID]; !ok {
                        c.clients[tuple.ID] = tuple.AppClient
                        if !c.noControls {
                                 tuple.AppClient.ControlConnection()
                        }
                }
```

But I don't think it's the right thing to do, since, if called by error, `PipeConnection()` will try to contact the app.